packages/dds/map/src/interfaces.ts

```diff
@@ -346,11 +348,14 @@ export interface ISharedMapEvents extends ISharedObjectEvents {
  *
  * For more information, including example usages, see {@link https://fluidframework.com/docs/data-structures/map/}.
  * @sealed
- * @alpha
+ * @beta
```

Comment:
I do not fully understand why we downgrade it to beta, and what problem it solves.
I think (from description and other changes) you are breaking dependencies between ISharedMap and such things as IChannel, but I'm not sure how it's connected to tag change

Reply:
Like everything else in this PR moving from alpha to beta is intended as a step toward becoming public. If we went all the way to public these API would stop being under only fluid-framework/legacy which is where we want them.

I think the important thins are getting things in a state where the type dependencies make these possible to become public, and keeping these APIs under fluid-framework/legacy not just fluid-framework. This accomplishes both of those.

If we want to block merging this until we have done the needed work to go all the way to public without ending up in  fluid-framework, thats fine, and we can evaluate the actual content of this PR while waiting for that.

> I think (from description and other changes) you are breaking dependencies between ISharedMap and such things as IChannel, but I'm not sure how it's connected to tag change

Without the tag change, there is no tooling based check that this decoupling was done correctly. With this change (and that CI is passing) we have ensured that we can mark these things public without having to pull back in things like IFluidDataStoreRuntime, IChannel etc.. All the work in this PR started from making the tag change, then fixing the errors that causes (including more tag changes, and adjusting dependencies, cleaning up tests etc.) 

At least while this PR is in the draft stage and subject to significant changes, keeping the tag change is very useful to ensuring we didn't miss any problematic type dependencies, but it would be reasonable to move the tags back to alpha or all the way to public before merging depending on how we want to stage things.